From 376859ee5b6b28ad6a8a535dee091b8b045b05b4 Mon Sep 17 00:00:00 2001
From: Juergen Borleis <jbe@pengutronix.de>
Date: Fri, 20 Nov 2015 14:57:57 +0100
Subject: [PATCH 10/15] blspec: Prepare for a different compatible string

In order to select a spec entry corresponding to run-time detected
information, prepare the section routine to use a different 'compatible'
string to search for.

Signed-off-by: Juergen Borleis <jbe@pengutronix.de>
---
 common/blspec.c | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/common/blspec.c b/common/blspec.c
index d5ddb4fbbddc..1000863ad3e8 100644
--- a/common/blspec.c
+++ b/common/blspec.c
@@ -250,17 +250,19 @@ static bool entry_is_of_compatible(struct blspec_entry *entry)
 	void *fdt = NULL;
 	int ret;
 	struct device_node *root = NULL, *barebox_root;
-	const char *compat;
+	const char *compat = NULL;
 	char *filename;
 
-	/* If we don't have a root node every entry is compatible */
-	barebox_root = of_get_root_node();
-	if (!barebox_root)
-		return true;
+	if (compat == NULL) {
+		/* If we don't have a root node every entry is compatible */
+		barebox_root = of_get_root_node();
+		if (!barebox_root)
+			return true;
 
-	ret = of_property_read_string(barebox_root, "compatible", &compat);
-	if (ret)
-		return false;
+		ret = of_property_read_string(barebox_root, "compatible", &compat);
+		if (ret)
+			return false;
+	}
 
 	if (entry->rootpath)
 		abspath = entry->rootpath;
-- 
2.1.4

